subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -251,6 +254,82 @@ int nrf_cloud_decode_requested_state(const struct nrf_cloud_data *input,
 	return 0;
 }
 
+int nrf_cloud_encode_config_response(struct nrf_cloud_data const *const input,
+				     struct nrf_cloud_data *const output,
+				     bool *const has_config)
+{
+	__ASSERT_NO_MSG(output != NULL);
+	char *buffer;
+	cJSON *root_obj = cJSON_CreateObject();
```

Comment:
We pre-allocate `root_obj`, `desired_obj` etc., yet they are only used if `(state_obj && config_obj)` condition is satisfied. So we may end up allocating them, just to have them freed unused. Would it make sense to:
1) Parse `input` first,
2) Exit early in case no response is neeed,
3) Otherwise proceed with allocations and further logic.

IMO that would make the function easier to follow.